// @universal-i18n/react/dist/index.js
"use strict";
"use client";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  ALL_LOCALES: () => ALL_LOCALES,
  AutoTranslateProvider: () => AutoTranslateProvider,
  useAutoTranslate: () => useAutoTranslate
});
module.exports = __toCommonJS(src_exports);

// src/client.tsx
var import_react = require("react");
var import_jsx_runtime = require("react/jsx-runtime");
var DEFAULT_SKIP_SELECTORS = "script, style, code, pre, kbd, samp, var, svg, math, [data-i18n-skip], [contenteditable], input, textarea, select";
var LOCALE_LABELS = {
  // Top Tier
  en: "\u{1F1FA}\u{1F1F8} English",
  es: "\u{1F1EA}\u{1F1F8} Espa\xF1ol",
  fr: "\u{1F1EB}\u{1F1F7} Fran\xE7ais",
  de: "\u{1F1E9}\u{1F1EA} Deutsch",
  it: "\u{1F1EE}\u{1F1F9} Italiano",
  pt: "\u{1F1E7}\u{1F1F7} Portugu\xEAs",
  ja: "\u{1F1EF}\u{1F1F5} \u65E5\u672C\u8A9E",
  ko: "\u{1F1F0}\u{1F1F7} \uD55C\uAD6D\uC5B4",
  zh: "\u{1F1E8}\u{1F1F3} \u4E2D\u6587",
  ar: "\u{1F1F8}\u{1F1E6} \u0627\u0644\u0639\u0631\u0628\u064A\u0629",
  hi: "\u{1F1EE}\u{1F1F3} \u0939\u093F\u0928\u094D\u0926\u0940",
  ru: "\u{1F1F7}\u{1F1FA} \u0420\u0443\u0441\u0441\u043A\u0438\u0439",
  // Western Europe & Nordics
  nl: "\u{1F1F3}\u{1F1F1} Nederlands",
  sv: "\u{1F1F8}\u{1F1EA} Svenska",
  da: "\u{1F1E9}\u{1F1F0} Dansk",
  fi: "\u{1F1EB}\u{1F1EE} Suomi",
  nb: "\u{1F1F3}\u{1F1F4} Norsk",
  is: "\u{1F1EE}\u{1F1F8} \xCDslenska",
  ga: "\u{1F1EE}\u{1F1EA} Gaeilge",
  cy: "\u{1F3F4}\u{E0067}\u{E0062}\u{E0077}\u{E006C}\u{E0073}\u{E007F} Cymraeg",
  gl: "\u{1F1EA}\u{1F1F8} Galego",
  eu: "\u{1F1EA}\u{1F1F8} Euskara",
  ca: "\u{1F1EA}\u{1F1F8} Catal\xE0",
  // Eastern Europe & Balkans
  pl: "\u{1F1F5}\u{1F1F1} Polski",
  cs: "\u{1F1E8}\u{1F1FF} \u010Ce\u0161tina",
  sk: "\u{1F1F8}\u{1F1F0} Sloven\u010Dina",
  hu: "\u{1F1ED}\u{1F1FA} Magyar",
  ro: "\u{1F1F7}\u{1F1F4} Rom\xE2n\u0103",
  bg: "\u{1F1E7}\u{1F1EC} \u0411\u044A\u043B\u0433\u0430\u0440\u0441\u043A\u0438",
  sr: "\u{1F1F7}\u{1F1F8} \u0421\u0440\u043F\u0441\u043A\u0438",
  hr: "\u{1F1ED}\u{1F1F7} Hrvatski",
  bs: "\u{1F1E7}\u{1F1E6} Bosanski",
  sl: "\u{1F1F8}\u{1F1EE} Sloven\u0161\u010Dina",
  el: "\u{1F1EC}\u{1F1F7} \u0395\u03BB\u03BB\u03B7\u03BD\u03B9\u03BA\u03AC",
  tr: "\u{1F1F9}\u{1F1F7} T\xFCrk\xE7e",
  uk: "\u{1F1FA}\u{1F1E6} \u0423\u043A\u0440\u0430\u0457\u043D\u0441\u044C\u043A\u0430",
  be: "\u{1F1E7}\u{1F1FE} \u0411\u0435\u043B\u0430\u0440\u0443\u0441\u043A\u0430\u044F",
  mk: "\u{1F1F2}\u{1F1F0} \u041C\u0430\u043A\u0435\u0434\u043E\u043D\u0441\u043A\u0438",
  sq: "\u{1F1E6}\u{1F1F1} Shqip",
  // Asia Pacific
  th: "\u{1F1F9}\u{1F1ED} \u0E44\u0E17\u0E22",
  vi: "\u{1F1FB}\u{1F1F3} Ti\u1EBFng Vi\u1EC7t",
  id: "\u{1F1EE}\u{1F1E9} Bahasa Indonesia",
  ms: "\u{1F1F2}\u{1F1FE} Bahasa Melayu",
  tl: "\u{1F1F5}\u{1F1ED} Tagalog",
  km: "\u{1F1F0}\u{1F1ED} \u1781\u17D2\u1798\u17C2\u179A",
  lo: "\u{1F1F1}\u{1F1E6} \u0EA5\u0EB2\u0EA7",
  my: "\u{1F1F2}\u{1F1F2} \u1019\u103C\u1014\u103A\u1019\u102C",
  mn: "\u{1F1F2}\u{1F1F3} \u041C\u043E\u043D\u0433\u043E\u043B",
  // South Asia
  bn: "\u{1F1E7}\u{1F1E9} \u09AC\u09BE\u0982\u09B2\u09BE",
  ta: "\u{1F1EE}\u{1F1F3} \u0BA4\u0BAE\u0BBF\u0BB4\u0BCD",
  te: "\u{1F1EE}\u{1F1F3} \u0C24\u0C46\u0C32\u0C41\u0C17\u0C41",
  mr: "\u{1F1EE}\u{1F1F3} \u092E\u0930\u093E\u0920\u0940",
  gu: "\u{1F1EE}\u{1F1F3} \u0A97\u0AC1\u0A9C\u0AB0\u0ABE\u0AA4\u0AC0",
  kn: "\u{1F1EE}\u{1F1F3} \u0C95\u0CA8\u0CCD\u0CA8\u0CA1",
  ml: "\u{1F1EE}\u{1F1F3} \u0D2E\u0D32\u0D2F\u0D3E\u0D33\u0D02",
  pa: "\u{1F1EE}\u{1F1F3} \u0A2A\u0A70\u0A1C\u0A3E\u0A2C\u0A40",
  ur: "\u{1F1F5}\u{1F1F0} \u0627\u0631\u062F\u0648",
  si: "\u{1F1F1}\u{1F1F0} \u0DC3\u0DD2\u0D82\u0DC4\u0DBD",
  ne: "\u{1F1F3}\u{1F1F5} \u0928\u0947\u092A\u093E\u0932\u0940",
  "zh-TW": "\u{1F1F9}\u{1F1FC} \u7E41\u9AD4\u4E2D\u6587",
  "zh-CN": "\u{1F1E8}\u{1F1F3} \u7B80\u4F53\u4E2D\u6587",
  // Middle East & Central Asia
  he: "\u{1F1EE}\u{1F1F1} \u05E2\u05D1\u05E8\u05D9\u05EA",
  fa: "\u{1F1EE}\u{1F1F7} \u0641\u0627\u0631\u0633\u06CC",
  ps: "\u{1F1E6}\u{1F1EB} \u067E\u069A\u062A\u0648",
  ku: "\u{1F1EE}\u{1F1F6} Kurd\xEE",
  az: "\u{1F1E6}\u{1F1FF} Az\u0259rbaycan",
  ka: "\u{1F1EC}\u{1F1EA} \u10E5\u10D0\u10E0\u10D7\u10E3\u10DA\u10D8",
  hy: "\u{1F1E6}\u{1F1F2} \u0540\u0561\u0575\u0565\u0580\u0565\u0576",
  uz: "\u{1F1FA}\u{1F1FF} O\u02BBzbek",
  kk: "\u{1F1F0}\u{1F1FF} \u049A\u0430\u0437\u0430\u049B",
  ky: "\u{1F1F0}\u{1F1EC} \u041A\u044B\u0440\u0433\u044B\u0437",
  tg: "\u{1F1F9}\u{1F1EF} \u0422\u043E\u04B7\u0438\u043A\u04E3",
  tk: "\u{1F1F9}\u{1F1F2} T\xFCrkmen",
  // Africa
  sw: "\u{1F1F0}\u{1F1EA} Kiswahili",
  am: "\u{1F1EA}\u{1F1F9} \u12A0\u121B\u122D\u129B",
  yo: "\u{1F1F3}\u{1F1EC} Yor\xF9b\xE1",
  ig: "\u{1F1F3}\u{1F1EC} Igbo",
  ha: "\u{1F1F3}\u{1F1EC} Hausa",
  zu: "\u{1F1FF}\u{1F1E6} isiZulu",
  xh: "\u{1F1FF}\u{1F1E6} isiXhosa",
  af: "\u{1F1FF}\u{1F1E6} Afrikaans",
  sn: "\u{1F1FF}\u{1F1FC} chiShona",
  mg: "\u{1F1F2}\u{1F1EC} Malagasy",
  so: "\u{1F1F8}\u{1F1F4} Soomaali",
  // Baltics
  et: "\u{1F1EA}\u{1F1EA} Eesti",
  lv: "\u{1F1F1}\u{1F1FB} Latvie\u0161u",
  lt: "\u{1F1F1}\u{1F1F9} Lietuvi\u0173",
  // Other Global
  mt: "\u{1F1F2}\u{1F1F9} Malti",
  eo: "\u{1F30D} Esperanto",
  la: "\u{1F3DB}\uFE0F Latina"
};
var ALL_LOCALES = Object.keys(LOCALE_LABELS);
var AutoTranslateContext = (0, import_react.createContext)({
  locale: "en",
  sourceLocale: "en",
  setLocale: () => {
  },
  availableLocales: ["en"],
  isTranslating: false
});
function useAutoTranslate() {
  return (0, import_react.useContext)(AutoTranslateContext);
}
var DOMTranslationEngine = class {
  constructor(apiKey, sourceLocale, targetLocale, batchDelayMs, apiRoute, fetchAction) {
    this.apiKey = apiKey;
    this.sourceLocale = sourceLocale;
    this.targetLocale = targetLocale;
    this.batchDelayMs = batchDelayMs;
    this.apiRoute = apiRoute;
    this.fetchAction = fetchAction;
    this.loadCacheFromStorage();
  }
  cache = {};
  pendingTexts = /* @__PURE__ */ new Map();
  batchTimer = null;
  loadCacheFromStorage() {
    if (typeof window === "undefined") return;
    try {
      const saved = localStorage.getItem(
        `ui18n-cache-${this.sourceLocale}-${this.targetLocale}`
      );
      if (saved) {
        this.cache = JSON.parse(saved);
      }
    } catch {
    }
  }
  saveCacheToStorage() {
    if (typeof window === "undefined") return;
    try {
      localStorage.setItem(
        `ui18n-cache-${this.sourceLocale}-${this.targetLocale}`,
        JSON.stringify(this.cache)
      );
    } catch {
    }
  }
  getCacheKey(text) {
    return text.trim();
  }
  getCachedTranslation(text) {
    const key = this.getCacheKey(text);
    return this.cache[key] ?? null;
  }
  async translate(text) {
    const trimmed = text.trim();
    if (!trimmed || trimmed.length < 2) return text;
    const cached = this.cache[this.getCacheKey(trimmed)];
    if (cached) return cached;
    return new Promise((resolve) => {
      const existing = this.pendingTexts.get(trimmed);
      if (existing) {
        existing.push(resolve);
      } else {
        this.pendingTexts.set(trimmed, [resolve]);
      }
      if (!this.batchTimer) {
        this.batchTimer = setTimeout(
          () => this.flushBatch(),
          this.batchDelayMs
        );
      }
    });
  }
  async flushBatch() {
    this.batchTimer = null;
    const batch = new Map(this.pendingTexts);
    this.pendingTexts.clear();
    if (batch.size === 0) return;
    const texts = Array.from(batch.keys());
    try {
      const inputObj = {};
      texts.forEach((text, i) => {
        inputObj[`k${i}`] = text;
      });
      let translated = {};
      if (this.fetchAction) {
        translated = await this.fetchAction(
          inputObj,
          this.sourceLocale,
          this.targetLocale
        );
      } else if (this.apiRoute || !this.apiKey && typeof window !== "undefined") {
        const route = this.apiRoute || "/api/universal-i18n";
        const response = await fetch(route, {
          method: "POST",
          headers: { "Content-Type": "application/json" },
          body: JSON.stringify({
            texts: inputObj,
            source: this.sourceLocale,
            target: this.targetLocale
          })
        });
        if (!response.ok) {
          throw new Error(`Proxy API error: ${response.status}`);
        }
        translated = await response.json();
      } else {
        const endpoint = "https://engine.lingo.dev/i18n";
        const headers = {
          "Content-Type": "application/json; charset=utf-8"
        };
        if (this.apiKey) {
          headers["Authorization"] = `Bearer ${this.apiKey}`;
        }
        const response = await fetch(endpoint, {
          method: "POST",
          headers,
          body: JSON.stringify({
            params: { workflowId: Math.random().toString(36).substring(7) },
            locale: { source: this.sourceLocale, target: this.targetLocale },
            data: inputObj
          })
        });
        if (!response.ok) {
          throw new Error(`API error: ${response.status}`);
        }
        const responseData = await response.json();
        if (responseData.error && !responseData.data) {
          throw new Error(responseData.error);
        }
        translated = responseData.data || {};
      }
      texts.forEach((originalText, i) => {
        const translatedText = translated[`k${i}`] || originalText;
        this.cache[this.getCacheKey(originalText)] = translatedText;
        const resolvers = batch.get(originalText);
        if (resolvers) {
          resolvers.forEach((resolve) => resolve(translatedText));
        }
      });
      this.saveCacheToStorage();
    } catch (error) {
      console.error("[universal-i18n] Batch failed:", error);
      for (const [originalText, resolvers] of Array.from(batch.entries())) {
        resolvers.forEach((resolve) => resolve(originalText));
      }
    }
  }
  clearCache() {
    this.cache = {};
    if (typeof window !== "undefined") {
      localStorage.removeItem(
        `ui18n-cache-${this.sourceLocale}-${this.targetLocale}`
      );
    }
  }
};
function getTextNodes(root, skipSelector) {
  const textNodes = [];
  const walker = document.createTreeWalker(root, NodeFilter.SHOW_TEXT, {
    acceptNode(node2) {
      if (!node2.textContent || !node2.textContent.trim()) {
        return NodeFilter.FILTER_REJECT;
      }
      if (node2.textContent.trim().length < 2) {
        return NodeFilter.FILTER_REJECT;
      }
      const parent = node2.parentElement;
      if (!parent) return NodeFilter.FILTER_REJECT;
      if (parent.closest(skipSelector)) {
        return NodeFilter.FILTER_REJECT;
      }
      if (parent.hasAttribute("data-i18n-translated")) {
        return NodeFilter.FILTER_REJECT;
      }
      return NodeFilter.FILTER_ACCEPT;
    }
  });
  let node;
  while (node = walker.nextNode()) {
    textNodes.push(node);
  }
  return textNodes;
}
function AutoTranslateProvider({
  apiKey,
  sourceLocale = "en",
  targetLocale: initialTargetLocale,
  availableLocales = "all",
  batchDelayMs = 150,
  skipSelectors = DEFAULT_SKIP_SELECTORS,
  showSwitcher = true,
  onLocaleChange,
  apiRoute,
  fetchAction,
  children
}) {
  const resolvedLocales = availableLocales === "all" ? Object.keys(LOCALE_LABELS) : availableLocales || ["en"];
  const [locale, setLocaleState] = (0, import_react.useState)(() => {
    if (typeof window !== "undefined") {
      const saved = localStorage.getItem("ui18n-auto-locale");
      if (saved && resolvedLocales.includes(saved)) return saved;
    }
    if (!initialTargetLocale && typeof navigator !== "undefined") {
      const browserLang = navigator.language?.split("-")[0];
      if (browserLang && browserLang !== sourceLocale && resolvedLocales.includes(browserLang)) {
        return browserLang;
      }
    }
    return initialTargetLocale || sourceLocale;
  });
  const [isTranslating, setIsTranslating] = (0, import_react.useState)(false);
  const engineRef = (0, import_react.useRef)(null);
  const observerRef = (0, import_react.useRef)(null);
  const containerRef = (0, import_react.useRef)(null);
  const originalTextsRef = (0, import_react.useRef)(/* @__PURE__ */ new Map());
  const resolvedApiKey = apiKey || "";
  (0, import_react.useEffect)(() => {
    if (locale === sourceLocale) {
      engineRef.current = null;
      return;
    }
    engineRef.current = new DOMTranslationEngine(
      resolvedApiKey,
      sourceLocale,
      locale,
      batchDelayMs,
      apiRoute,
      fetchAction
    );
  }, [
    locale,
    sourceLocale,
    resolvedApiKey,
    batchDelayMs,
    apiRoute,
    fetchAction
  ]);
  const translateDOM = (0, import_react.useCallback)(async () => {
    const engine = engineRef.current;
    const container = containerRef.current;
    if (!engine || !container || locale === sourceLocale) return;
    setIsTranslating(true);
    try {
      const textNodes = getTextNodes(container, skipSelectors);
      if (textNodes.length === 0) {
        setIsTranslating(false);
        return;
      }
      const toTranslate = [];
      for (const node of textNodes) {
        const originalText = node.textContent || "";
        if (!originalTextsRef.current.has(node)) {
          originalTextsRef.current.set(node, originalText);
        }
        const cached = engine.getCachedTranslation(originalText.trim());
        if (cached) {
          node.textContent = originalText.replace(originalText.trim(), cached);
          const parent = node.parentElement;
          if (parent) parent.setAttribute("data-i18n-translated", "true");
        } else {
          toTranslate.push({ node, text: originalText });
        }
      }
      if (toTranslate.length > 0) {
        const translations = await Promise.all(
          toTranslate.map(({ text }) => engine.translate(text))
        );
        translations.forEach((translated, i) => {
          const { node, text: originalText } = toTranslate[i];
          if (node.parentElement) {
            node.textContent = originalText.replace(
              originalText.trim(),
              translated
            );
            node.parentElement.setAttribute("data-i18n-translated", "true");
          }
        });
      }
    } catch (error) {
      console.error("[universal-i18n] DOM translation error:", error);
    } finally {
      setIsTranslating(false);
    }
  }, [locale, sourceLocale, skipSelectors]);
  const revertDOM = (0, import_react.useCallback)(() => {
    for (const [node, originalText] of Array.from(
      originalTextsRef.current.entries()
    )) {
      if (node.parentElement) {
        node.textContent = originalText;
        node.parentElement.removeAttribute(
          "data-i18n-translated"
        );
      }
    }
    originalTextsRef.current.clear();
  }, []);
  (0, import_react.useEffect)(() => {
    const container = containerRef.current;
    if (!container) return;
    const timer = setTimeout(() => translateDOM(), 100);
    observerRef.current = new MutationObserver((mutations) => {
      let hasNewText = false;
      for (const mutation of mutations) {
        if (mutation.type === "childList" && mutation.addedNodes.length > 0) {
          for (const node of Array.from(mutation.addedNodes)) {
            if (node.nodeType === Node.ELEMENT_NODE || node.nodeType === Node.TEXT_NODE) {
              hasNewText = true;
              break;
            }
          }
        }
        if (hasNewText) break;
      }
      if (hasNewText) {
        setTimeout(() => translateDOM(), 200);
      }
    });
    observerRef.current.observe(container, {
      childList: true,
      subtree: true
    });
    return () => {
      clearTimeout(timer);
      observerRef.current?.disconnect();
    };
  }, [translateDOM]);
  const setLocale = (0, import_react.useCallback)(
    (newLocale) => {
      revertDOM();
      setLocaleState(newLocale);
      if (typeof window !== "undefined") {
        localStorage.setItem("ui18n-auto-locale", newLocale);
      }
      onLocaleChange?.(newLocale);
    },
    [revertDOM, onLocaleChange]
  );
  (0, import_react.useEffect)(() => {
    if (locale !== sourceLocale) {
      const timer = setTimeout(() => translateDOM(), 150);
      return () => clearTimeout(timer);
    }
  }, [locale, sourceLocale, translateDOM]);
  const contextValue = {
    locale,
    sourceLocale,
    setLocale,
    availableLocales: resolvedLocales,
    isTranslating
  };
  return /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(AutoTranslateContext.Provider, { value: contextValue, children: [
    /* @__PURE__ */ (0, import_jsx_runtime.jsx)("div", { ref: containerRef, "data-i18n-root": "true", children }),
    showSwitcher && resolvedLocales.length > 1 && /* @__PURE__ */ (0, import_jsx_runtime.jsx)(FloatingLanguageSwitcher, {}),
    isTranslating && /* @__PURE__ */ (0, import_jsx_runtime.jsx)(TranslatingIndicator, {})
  ] });
}
function FloatingLanguageSwitcher() {
  const { locale, setLocale, availableLocales } = useAutoTranslate();
  const [open, setOpen] = (0, import_react.useState)(false);
  const [mounted, setMounted] = (0, import_react.useState)(false);
  (0, import_react.useEffect)(() => {
    setMounted(true);
  }, []);
  if (!mounted) return null;
  return /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(
    "div",
    {
      style: {
        position: "fixed",
        bottom: "24px",
        right: "24px",
        zIndex: 9999,
        fontFamily: "system-ui, -apple-system, sans-serif"
      },
      children: [
        open && /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
          "div",
          {
            style: {
              position: "absolute",
              bottom: "56px",
              right: "0",
              background: "white",
              borderRadius: "16px",
              boxShadow: "0 20px 60px rgba(0,0,0,0.15), 0 0 0 1px rgba(0,0,0,0.05)",
              padding: "8px",
              minWidth: "180px",
              maxHeight: "320px",
              overflowY: "auto"
            },
            children: availableLocales.map((l) => /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
              "button",
              {
                onClick: () => {
                  setLocale(l);
                  setOpen(false);
                },
                style: {
                  display: "block",
                  width: "100%",
                  padding: "10px 16px",
                  border: "none",
                  background: l === locale ? "#EEF2FF" : "transparent",
                  borderRadius: "10px",
                  cursor: "pointer",
                  textAlign: "left",
                  fontSize: "14px",
                  fontWeight: l === locale ? "700" : "400",
                  color: l === locale ? "#4F46E5" : "#374151",
                  transition: "background 0.15s"
                },
                onMouseEnter: (e) => {
                  if (l !== locale)
                    e.target.style.background = "#F3F4F6";
                },
                onMouseLeave: (e) => {
                  if (l !== locale)
                    e.target.style.background = "transparent";
                },
                children: LOCALE_LABELS[l] || l.toUpperCase()
              },
              l
            ))
          }
        ),
        /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
          "button",
          {
            onClick: () => setOpen(!open),
            style: {
              width: "48px",
              height: "48px",
              borderRadius: "50%",
              border: "none",
              background: "linear-gradient(135deg, #6366F1, #8B5CF6)",
              color: "white",
              fontSize: "20px",
              cursor: "pointer",
              boxShadow: "0 8px 32px rgba(99,102,241,0.4), 0 0 0 4px rgba(99,102,241,0.1)",
              display: "flex",
              alignItems: "center",
              justifyContent: "center",
              transition: "transform 0.2s, box-shadow 0.2s"
            },
            onMouseEnter: (e) => {
              e.target.style.transform = "scale(1.1)";
            },
            onMouseLeave: (e) => {
              e.target.style.transform = "scale(1)";
            },
            title: `Current language: ${LOCALE_LABELS[locale] || locale}`,
            children: "\u{1F310}"
          }
        )
      ]
    }
  );
}
function TranslatingIndicator() {
  return /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(
    "div",
    {
      style: {
        position: "fixed",
        top: "16px",
        left: "50%",
        transform: "translateX(-50%)",
        zIndex: 1e4,
        background: "linear-gradient(135deg, #6366F1, #8B5CF6)",
        color: "white",
        padding: "8px 20px",
        borderRadius: "100px",
        fontSize: "13px",
        fontWeight: "600",
        fontFamily: "system-ui, -apple-system, sans-serif",
        boxShadow: "0 8px 32px rgba(99,102,241,0.3)",
        display: "flex",
        alignItems: "center",
        gap: "8px",
        animation: "ui18n-pulse 1.5s ease-in-out infinite"
      },
      children: [
        /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
          "span",
          {
            style: {
              width: "8px",
              height: "8px",
              borderRadius: "50%",
              background: "white",
              animation: "ui18n-dot 1s ease-in-out infinite"
            }
          }
        ),
        "Translating...",
        /* @__PURE__ */ (0, import_jsx_runtime.jsx)("style", { children: `
        @keyframes ui18n-pulse {
          0%, 100% { opacity: 1; }
          50% { opacity: 0.8; }
        }
        @keyframes ui18n-dot {
          0%, 100% { transform: scale(1); }
          50% { transform: scale(1.5); }
        }
      ` })
      ]
    }
  );
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ALL_LOCALES,
  AutoTranslateProvider,
  useAutoTranslate
});
