// @universal-i18n/react/dist/server.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/server.ts
var server_exports = {};
__export(server_exports, {
  createTranslationRoute: () => createTranslationRoute
});
module.exports = __toCommonJS(server_exports);
function createTranslationRoute() {
  return async function POST(req) {
    const apiKey = process.env.NEXT_PUBLIC_LINGODOTDEV_API_KEY || process.env.LINGODOTDEV_API_KEY;
    if (!apiKey) {
      console.error("[universal-i18n] Missing API key. Set LINGODOTDEV_API_KEY.");
      return new Response(JSON.stringify({ error: "Missing API key" }), {
        status: 500,
        headers: { "Content-Type": "application/json" }
      });
    }
    try {
      const { texts, source, target } = await req.json();
      if (!texts || !source || !target) {
        return new Response(JSON.stringify({ error: "Missing fields" }), {
          status: 400,
          headers: { "Content-Type": "application/json" }
        });
      }
      const response = await fetch("https://engine.lingo.dev/i18n", {
        method: "POST",
        headers: {
          "Content-Type": "application/json; charset=utf-8",
          "Authorization": `Bearer ${apiKey}`
        },
        body: JSON.stringify({
          params: { workflowId: Math.random().toString(36).substring(7) },
          locale: { source, target },
          data: texts
        })
      });
      if (!response.ok) {
        return new Response(JSON.stringify({ error: "Upstream Proxy Failed" }), {
          status: response.status,
          headers: { "Content-Type": "application/json" }
        });
      }
      const json = await response.json();
      return new Response(JSON.stringify(json.data || {}), {
        status: 200,
        headers: { "Content-Type": "application/json" }
      });
    } catch (err) {
      return new Response(JSON.stringify({ error: err.message }), {
        status: 500,
        headers: { "Content-Type": "application/json" }
      });
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createTranslationRoute
});
